File: 2015/01/main.go
package main

import (
	"fmt"
	"io/ioutil"
)

func main(){
	andar := 0
	posicao := 0
	subsolo := 0

	arquivo, err := ioutil.ReadFile("input.txt")

	if err != nil{
		fmt.Println("Erro ao abrir arquivo")
		return
	}

	for i := 0; i < len(arquivo); i++ {
		posicao = posicao + 1;
		char := string(arquivo[i])
		if char == "(" {
			andar = andar + 1;
		}else if char == ")" {
			andar = andar - 1;
		}

		if subsolo == 0 {
			if andar == -1 {
				subsolo = posicao
			}
		}
	}

	fmt.Println("Resultado parte 1: ", andar)
	fmt.Println("Resultado parte 2: ", subsolo)
}

File: 2015/01/Main.java
import java.io.*;

public class Main{
	public static void main(String args[]){
		int andar = 0;
		int subsolo = 0;

		char sobe = "(".charAt(0);
		char desce = ")".charAt(0);

		String arquivo = "input.txt";
		try{
			BufferedReader br = new BufferedReader(new FileReader(arquivo));
			String linhas = br.readLine();
			for(int i = 0; i < linhas.length(); i++){
				char letra = linhas.charAt(i);
				if(letra == sobe){
					andar = andar + 1;
				}else if(letra == desce){
					andar = andar - 1;
				}

				if(subsolo == 0){
					if(andar == -1){
						subsolo = i + 1;
					}
				}
			}
			System.out.println("Resultado parte 1: "+andar);
			System.out.println("Resultado parte 2: "+subsolo);
		}catch(FileNotFoundException ex){
			ex.printStackTrace();
		}catch(IOException e){
			e.printStackTrace();
		}
	}
}
